fuseki-docker/Dockerfile

```diff
@@ -0,0 +1,154 @@
+## Licensed to the Apache Software Foundation (ASF) under one or more
+## contributor license agreements.  See the NOTICE file distributed with
+## this work for additional information regarding copyright ownership.
+## The ASF licenses this file to You under the Apache License, Version 2.0
+## (the "License"); you may not use this file except in compliance with
+## the License.  You may obtain a copy of the License at
+##
+##     http://www.apache.org/licenses/LICENSE-2.0
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS,
+## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+## See the License for the specific language governing permissions and
+## limitations under the License.
+
+## Apache Jena Fuseki server Dockerfile.
+
+## This Dockefile builds a reduced footprint container.
+
+ARG JAVA_VERSION=17
+
+ARG ALPINE_VERSION=3.17.1
+ARG JENA_VERSION=""
+ARG LOCK_UNLOCK_VERSION=""
+
+# Internal, passed between stages.
+ARG FUSEKI_DIR=/fuseki
+ARG FUSEKI_JAR=lock-unlock-fuseki-${JENA_VERSION}-${LOCK_UNLOCK_VERSION}.jar
+ARG FUSEKI_WEBAPP_WAR=jena-fuseki-war-${JENA_VERSION}.war
+ARG JAVA_MINIMAL=/opt/java-minimal
+
+## ---- Stage: Build java.
+FROM eclipse-temurin:${JAVA_VERSION}-alpine AS build
+
+ARG LOCK_UNLOCK_VERSION
+ARG JENA_VERSION
+ARG FUSEKI_JAR
+
+RUN [ "${JENA_VERSION}" != "" ] || { echo -e '\n**** Set JENA_VERSION ****\n' ; exit 1 ; }
+RUN echo && echo "==== Docker build for Apache Jena Fuseki ${JENA_VERSION} ====" && echo
+
+# Alpine: For objcopy used in jlink
+RUN apk add --no-cache curl binutils maven
+
+## build lock-unlock fuseki fulljar
+WORKDIR /build
+COPY . .
+RUN mvn package
+
+## ---- Stage: Download and build java.
+FROM eclipse-temurin:${JAVA_VERSION}-alpine AS base
```

Comment:
De eerste twee stages kan je volgens mij samenvoegen. Ze gebruiken beide dezelfde image en introduceert nu vooral duplicate code, zonder dat dat nodig is. Zou je willen kijken of dat lukt?

Reply:
Dat was nu juist het probleem. Dat lukte niet :see_no_evil: Het punt is dat de `mvn package` een `.jar` file oplevert in de `/build/target` folder en die moet dan gekopieerd worden _binnen_ de Docker build. Ik kreeg dat niet voor elkaar ...

Oh! Maar d'r zit een voorbeeld verder in deze Dockerfile, niet? :smiley: I'll have a look!

Reply:
Got it! Will fix it ...

Reply:
Kan inderdaad met zoiets als:
```
# Alpine: For objcopy used in jlink
RUN apk add --no-cache curl binutils maven

## build lock-unlock fuseki fulljar
WORKDIR /app
COPY . .
RUN mvn package

## -- Workdir /fuseki.
WORKDIR $FUSEKI_DIR/webapp

RUN cp /app/target/${FUSEKI_JAR} .

# Fuseki dingen
```
🙂